Extend/Controller/Admin/PaymentMain.php

```diff
@@ -247,15 +280,16 @@ private function _validateRequestParameters($aParams)
         $sUser = $aParams['oxpayments__wdoxidee_httpuser'];
         $sPass = $aParams['oxpayments__wdoxidee_httppass'];
 
-        if ($sUrl && $sUser && $sPass) {
+        if ($this->_isCredentialsSet($sUrl, $sUser, $sPass)) {
             $oConfig = new Config($sUrl, $sUser, $sPass);
             $oTransactionService = $this->_getTransactionService($oConfig);
             return $oTransactionService->checkCredentials();
         }
 
         // handle the case of different configuration per currency (currently only Payolution)
         // there it's only checked if config values are entered on the frontend but no validation is done on the backend
-        return $this->_checkCurrencyConfigFields($aParams);
+        return $aParams['oxpayments__oxid'] === PayolutionInvoicePaymentMethod::getName(true)
```

Comment:
please, move the payment check to the method, and do an early `return true;`

Reply:
simplified the code